src/ios/SQLite.m

```diff
@@ -196,66 +201,81 @@ -(id) getDBPath:(NSString *)dbFile at:(NSString *)atkey {
         NSString *dblocation = options[@"dblocation"];
         if (dblocation == NULL) dblocation = @"nosync";
         NSLog(@"target database location: %@", dblocation);
-        
+
         dbname = [self getDBPath:dbfilename at:dblocation];
-        
+
         /* Option to create from resource (pre-populated) if db does not exist: */
         if (![[NSFileManager defaultManager] fileExistsAtPath:dbname] && assetFilePath != NULL) {
           NSLog(@"Copying pre-populated asset to the destination directory");
           [self createFromResource:assetFilePath withDbname:dbname];
         }
       }
-      
+
       NSLog(@"Opening db in mode %@, full path: %@", (sqlOpenFlags == SQLITE_OPEN_READONLY) ? @"READ ONLY" : @"READ_WRITE",dbname);
-      
+
       const char *name = [dbname UTF8String];
       sqlite3 *db;
-      
+
       if (sqlite3_open_v2(name, &db,sqlOpenFlags, NULL) != SQLITE_OK) {
         pluginResult = [SQLiteResult resultWithStatus:SQLiteStatus_ERROR messageAsString:@"Unable to open DB"];
         return;
       } else {
         sqlite3_create_function(db, "regexp", 2, SQLITE_ANY, NULL, &sqlite_regexp, NULL, NULL);
-        
+
         // for SQLCipher version:
         // NSString *dbkey = options[@"key"];
         // const char *key = NULL;
         // if (dbkey != NULL) key = [dbkey UTF8String];
         // if (key != NULL) sqlite3_key(db, key, strlen(key));
-        
+
         // Attempt to read the SQLite master table [to support SQLCipher version]:
         if(sqlite3_exec(db, (const char*)"SELECT count(*) FROM sqlite_master;", NULL, NULL, NULL) == SQLITE_OK) {
           NSValue *dbPointer = [NSValue valueWithPointer:db];
           openDBs[dbfilename] = @{ @"dbPointer": dbPointer, @"dbPath" : dbname};
+          [openDBsFilePaths setValue:dbname forKey:dbfilename];
```

Comment:
Rather than creating new storage bucket for paths, can you incorporate new data elements into openDBs entries? Each entry already has a dbPoint, dbPath, you can add dbFilePath?